### DISCUSSION TITLE
Does this not support SDXL?

### DISCUSSION THREAD

**Discussion by ILI5Games** [lshqqytiger/stable-diffusion-webui-directml#307] [Q&A] [answered]
I have many SD 1.5 models running, but when SDXL model runs it shows updates to very nice images in the preview, but turns black at the last 1% every time. Hours lost and stuck on only SD1.5 for now. Any help would be great. Discord ready for chat if you can help!

**Comment by pw405** [comment]
Does the command prompt show any error messages after it is done generating an image?

May be related to the arguments you have in your Webui-user.bat. 

What GPU are you running?

Also, you may need to update some stuff in your python venv (Virtual Environment) to run SDXL.  

You can copy/paste, then edit the text below to your Stable Diffusion folder path (whatever that is)  save as a .bat file using Notepad or any other text editor.  

I saved my script in the same folder as the Webui-user.bat.  Double click to run it and it should get everything updated for you. (Don't have Stable Diffusion running when you do). 

Let me know if it works!

```
cd C:\Users\username\STABLEDIFFUSION\WEBUI\stable-diffusion-webui-directml
git pull

cd venv\Scripts
call activate.bat

cd ..\..
pip install -r requirements.txt --upgrade

pip install diffusers --upgrade

pip install invisible_watermark transformers accelerate safetensors

pause
```

**Reply by ILI5Games** [reply]
No errors after running on UI or CMD. It just saves the image to the same place as if nothing is wrong. I thought it might have been an upscaler or refiner problem but when disabling them I have same issues.

Arguments are:
@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS=--opt-split-attention --skip-torch-cuda-test --disable-nan-check --medvram --theme dark --autolaunch

call webui.bat

GPU AMD Radeon RX 6800M 12GB VRAM
CPU AMD Ryzen 9 5980HX
64GB RAM

Below image shows what CMD shows for loading and use of SDXL then loading and use of SD1 or 1.5. Only difference I see is an extra progress line for 1.5 version which I am not sure about, but shouldn't be an issue as that is the one that works.

![Screenshot 2023-10-21 014102](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/99750544/d206a68f-82fc-47b3-b138-5df56d0a95fb)

Then tried the code you wrote and it seems like nothing works now. UI opens but now none of the models work. For every model now ui and cmd Says;

AttributeError: 'LatentDiffusion' object has no attribute 'is_onnx'

**Reply by ILI5Games** [reply]
Not sure what to do at this point? Did the script update something in the file that I can roll back or did my python/pip update on my computer to something I can't use?

Should I just redownload [stable-diffusion-webui-directml](https://github.com/lshqqytiger/stable-diffusion-webui-directml)

**Reply by ILI5Games** [reply]
__
        raise AttributeError("'{}' object has no attribute '{}'".format(
    AttributeError: 'LatentDiffusion' object has no attribute 'is_onnx'

---

**Reply by ILI5Games** [reply, marked as answer]
Deleted and reinstalled whole directory. It now runs again and after removing all arguments I think either --skip-torch-cuda-test --disable-nan-check were the cause of the issue. Now runs all models I throw at it. Still would like to use those if possible, but apparently it causes an issue so I'll leave it for now.

**Reply by pw405** [reply]
Sorry for just getting back, the `is_onnx` was a bug that affected EVERYBODY if you were downloading the latest repo.  See issue here:
https://github.com/lshqqytiger/stable-diffusion-webui-directml/issues/308

So, I'm on RDNA 3, and I know there are a few differences with RDNA 2 on startup commands.  

I personally run the following:  ` --medvram --no-half --upcast-sampling --precision full  --skip-torch-cuda-test`

I'm not sure, but I think the disable nan check sounds familiar for causing issues.  I can't recall exactly.  Skip Torch Cuda test simply eliminates the error message at startup.  CUDA is Nvidia exclusive language for their GPU's, and anybody on AMD will of course not have CUDA.  

Eventually, AMD will release ROCm, which is kinna like their version of CUDA, but at the moment the only way to use ROCm for Stable Diffusion is on Linux.

**Reply by ILI5Games** [reply]
Tried with your arguments and my computer doesn't like it. Weird how a generation apart can make a change like that. Works best for me with only  --medvram --autolaunch

Thanks for the help though! And good to know that problem was not just me. I was scared for a second.
